source/common/upstream/cluster_manager_impl.cc

```diff
@@ -1133,22 +1143,22 @@ ClusterManagerImpl::ThreadLocalClusterManagerImpl::ClusterEntry::connPool(
     return nullptr;
   }
 
-  // Inherit socket options from downstream connection, if set.
   std::vector<uint8_t> hash_key = {uint8_t(protocol)};
 
-  // Use downstream connection socket options for computing connection pool hash key, if any.
+  Network::Socket::OptionsSharedPtr upstream_options(std::make_shared<Network::Socket::Options>());
+  if (context) {
+    // Inherit socket options from downstream connection, if set.
```

Comment:
I guess this is related to my previous comment. It seems like this is an implementation detail that maybe we shouldn't discuss in the docs for the filter API since we might make this downstream merging optional later?

Reply:
That makes sense. All we're guaranteeing with the filter API call is that the upstream request will have the provided options. We note that that may impact the number of connection pools, but we probably shouldn't require that the implementation of the API ensure that downstream options are applied.